src/apps/backend/modules/password_reset_token/password_reset_token_service.py

```diff
@@ -0,0 +1,103 @@
+        account = AccountService.get_account_by_username(username=params.username)
+        token = PasswordResetTokenUtil.generate_password_reset_token()
+        password_reset_token = PasswordResetTokenWriter.create_password_reset_token(
+            account.id, token
+        )
+        PasswordResetTokenService.send_password_reset_email(
+            account.id, account.first_name, account.username, token
+        )
+        password_reset_token_dict = {
+            "id": str(password_reset_token.id),
+            "account": account.id,
+            "token": token,
+            "is_used": password_reset_token.is_used,
+            "is_expired": PasswordResetTokenUtil.is_token_expired(password_reset_token.expires_at),
+            "expires_at": password_reset_token.expires_at,
+        }
+
+        return PasswordResetToken(
+            id=password_reset_token_dict.get("id"),
+            account=password_reset_token_dict.get("account"),
+            token=password_reset_token_dict.get("token"),
+            is_used=password_reset_token_dict.get("is_used"),
+            is_expired=password_reset_token_dict.get("is_expired"),
+            expires_at=password_reset_token_dict.get("expires_at"),
+        )
+    @staticmethod
+    def get_password_reset_token_by_account_id(account_id: str) -> PasswordResetToken:
+        return PasswordResetTokenReader.get_password_reset_token_by_account_id(account_id)
+
+    @staticmethod
+    def set_password_reset_token_as_used_by_id(password_reset_token_id: str) -> PasswordResetToken:
+        return PasswordResetTokenWriter.set_password_reset_token_as_used(password_reset_token_id)
+
+    @staticmethod
+    def verify_password_reset_token(account_id: str, token: str) -> PasswordResetToken:
+        password_reset_token = PasswordResetTokenService.get_password_reset_token_by_account_id(account_id)
+
+        if password_reset_token.is_expired:
+            raise AccountBadRequestError(
+                f"Password reset link is expired for accountId {account_id}. Please retry with new link"
+            )
+        if password_reset_token.is_used:
+            raise AccountBadRequestError(
+                f"Password reset is already used for accountId {account_id}. Please retry with new link"
+            )
+
+        is_token_valid = PasswordResetTokenUtil.compare_password(
+            password=token,
+            hashed_password=password_reset_token.token
+        )
+        if not is_token_valid:
+            raise AccountBadRequestError(
+                f"Password reset link is invalid for accountId {account_id}. Please retry with new link."
+            )
+
+        return password_reset_token
+
+    @staticmethod
+    def send_password_reset_email(
+        account_id: str, 
+        first_name: str, 
+        username: str, 
+        password_reset_token: str
+    ) -> None:
+
+        if not ConfigService.get_accounts_config().get("password_reset_token_email_enabled"):
```

Comment:
Hmm, what's the use of it? Why do we need a feature flag for resetting password?

Reply:
It was also used in the MERN boilerplate so I used it here too. Should I skip this part?

Reply:
Please skip this part. Don't think feature flag for account reset is required.